File: java/benchmark/src/main/java/de/hhu/bsinfo/observatory/benchmark/result/Measurement.java
package de.hhu.bsinfo.observatory.benchmark.result;

public abstract class Measurement implements Cloneable {

    private final int operationCount;
    private final int operationSize;

    private long totalData;

    Measurement(int operationCount, int operationSize) {
        this.operationCount = operationCount;
        this.operationSize = operationSize;
        totalData = (long) operationCount * (long) operationSize;
    }

    public int getOperationCount() {
        return operationCount;
    }

    public int getOperationSize() {
        return operationSize;
    }

    public long getTotalData() {
        return totalData;
    }

    public abstract double getTotalTime();

    public void setTotalData(long totalData) {
        this.totalData = totalData;
    }

    @Override
    public String toString() {
        return "Measurement {" +
            "\n\t" + ValueFormatter.formatValue("operationCount", operationCount) +
            ",\n\t" + ValueFormatter.formatValue("operationSize", operationSize, "Byte") +
            ",\n\t" + ValueFormatter.formatValue("totalData", totalData, "Byte") +
            "\n}";
    }
}

File: cpp/src/observatory/result/Measurement.cpp
#include <sstream>
#include <observatory/util/ValueFormatter.h>
#include "Measurement.h"

namespace Observatory {

Measurement::Measurement(uint32_t operationCount, uint32_t operationSize) :
        operationCount(operationCount),
        operationSize(operationSize) {}

uint32_t Measurement::getOperationCount() const {
    return operationCount;
}

uint32_t Measurement::getOperationSize() const {
    return operationSize;
}

uint64_t Measurement::getTotalData() const {
    return totalData;
}

void Measurement::setTotalData(uint64_t totalData) {
    this->totalData = totalData;
}

Measurement::operator std::string() const {
    std::ostringstream stream;

    stream << "Measurement {"
            << "\n\t" << ValueFormatter::formatValue("operationCount", operationCount)
            << ",\n\t" << ValueFormatter::formatValue("operationSize", operationSize, "Byte")
            << ",\n\t" << ValueFormatter::formatValue("totalData", totalData, "Byte")
            << "\n}";

    return stream.str();
}

}